include/ppconsul/consul.h

```diff
@@ -225,12 +225,22 @@ namespace ppconsul {
         template<class... Params, class = kwargs::enable_if_kwargs_t<Params...>>
         std::string makeQuery(const Params&... params) const
         {
-            return parameters::makeQuery(kw::dc = dataCenter(), kw::token = defaultToken(), params...);
+            // Exclude kw::token from query (should be submitted as a header)
+            return parameters::makeQuery(kw::dc = dataCenter(), params..., kw::token = "");
         }
 
-        Response<std::string> get_impl(http::Status& status, const std::string& paty, const std::string& query) const;
-        std::string put_impl(http::Status& status, const std::string& path, const std::string& query, const std::string& data) const;
-        std::string del_impl(http::Status& status, const std::string& path, const std::string& query) const;
+        template<class... Params, class = kwargs::enable_if_kwargs_t<Params...>>
+        http::RequestHeaders makeHeaders(const Params&... params) const
+        {
+            return { { http::Token_Header_Name, kwargs::get_opt(kw::token, defaultToken(), params...) } };
```

Comment:
Please don't use initializer lists in non-test code because they usually involves one extra copy of its elements (it might be optimized-out by the compiler but I would not rely on this). Just call `emplace` instead.

Reply:
Do you mean something like that?
```c++
auto header = http::RequestHeaders{};
headers.emplace(http::Token_Header_Name, kwargs::get_opt(kw::token, defaultToken(), params...));
return headers;
```

Reply:
Yes